### PR TITLE
handle-error: only support to handle query error

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve? 
only support handle query error to avoid misuse
### What is changed and how it works?
add a check when handle-error

### Check List 

Tests 
 - Integration test

### CODE REVIEW

File: syncer/handle_error.go [hunk 1]
```diff
@@ -33,10 +33,13 @@ func (s *Syncer) HandleError(ctx context.Context, req *pb.HandleWorkerErrorReque
 	pos := req.BinlogPos
 
 	if len(pos) == 0 {
-		startLocation := s.getErrLocation()
+		startLocation, isQueryEvent := s.getErrLocation()
 		if startLocation == nil {
 			return fmt.Errorf("source '%s' has no error", s.cfg.SourceID)
 		}
+		if !isQueryEvent {
+			return fmt.Errorf("only support to handle query error currently")
```
Comment (lance6716) [inline syncer/handle_error.go:41]: How about change to "handle DDL error" (query is unfamiliar to me), and provide some guidance to handle non-DDL error

**Review by lance6716 — COMMENTED** [review]
rest LGTM

**Comment by lance6716** [comment]
LGTM

**Review by csuzhangxc — APPROVED** [review]
LGTM

**Pull request merged**